canary/Models/Record.cs

```diff
@@ -221,25 +205,32 @@ public static (CauseCodes record, List<Dictionary<string, string>> issues) Check
             }
             catch (Exception e)
             {
-                if (e.Message != null && e.Message.Contains(";"))
-                {
-                    foreach (string er in e.Message.Split(";"))
-                    {
-                        Dictionary<string, string> entry = new Dictionary<string, string>();
-                        entry.Add("message", er.Replace("Parser:", "").Trim());
-                        entry.Add("severity", "error");
-                        entries.Add(entry);
-                    }
-                }
-                else if (e.Message != null)
+                entries = DecorateErrors(e);
+            }
+            return (record: newCauses, issues: entries);
+        }
+
+        public static List<Dictionary<string, string>> DecorateErrors(Exception e) {
+            List<Dictionary<string, string>> entries = new List<Dictionary<string, string>>();
+
+            if (e.Message != null && e.Message.Contains(";"))
+            {
+                foreach (string er in e.Message.Split(";"))
                 {
                     Dictionary<string, string> entry = new Dictionary<string, string>();
-                    entry.Add("message", e.Message.Replace("Parser:", "").Trim());
+                    entry.Add("message", er.Replace("Parser:", "").Trim());
                     entry.Add("severity", "error");
                     entries.Add(entry);
                 }
             }
-            return (record: newCauses, issues: entries);
+            else if (e.Message != null)
```

Comment:
I don't think this block is necessary. if the e.Message.Contains(";") clause is remove from the if above then the loop will just run once and have the same effect as this block. Unless I'm missing something?

Reply:
Yep you are correct, I've gone ahead and removed it.